From 5d972400f8e7de2ceabb1cd95271e6c7ffff8640 Mon Sep 17 00:00:00 2001
From: Rabeeh Khoury <rabeeh@solid-run.com>
Date: Wed, 8 Nov 2017 15:54:27 +0200
Subject: [PATCH 1/2] mvebu: a38x: ddr : properly check calibration results

'pup' variable is not set; therefore the pup_state[x][pup] is being checked in
a value that is outside the array (pup=5 in the failure that was observed).

'pup' variable needs to be iterated 0..tm->num_of_bus_per_interface.

Signed-off-by: Rabeeh Khoury <rabeeh@solid-run.com>
---
 drivers/ddr/marvell/a38x/ddr3_training_pbs.c |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/ddr/marvell/a38x/ddr3_training_pbs.c b/drivers/ddr/marvell/a38x/ddr3_training_pbs.c
index 2b4a58f..84ee006 100644
--- a/drivers/ddr/marvell/a38x/ddr3_training_pbs.c
+++ b/drivers/ddr/marvell/a38x/ddr3_training_pbs.c
@@ -882,8 +882,9 @@ int ddr3_tip_pbs(u32 dev_num, enum pbs_dir pbs_mode)
		 * meaning that there is no VW exist at all (No lock at
		 * the EBA ADLL shift at EBS)
		 */
-		if (pup_state[if_id][pup] == 1)
-			return MV_FAIL;
+		for (pup = 0; pup < tm->num_of_bus_per_interface; pup++)
+			if (pup_state[if_id][pup] == 1)
+				return MV_FAIL;
	}

	return MV_OK;
--
1.7.9.5
